internal/serverinstall/k8s.go

```diff
@@ -379,139 +337,127 @@ func (i *K8sInstaller) Upgrade(
-	if podList, err := podClient.List(ctx, metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", serverName)}); err != nil {
-		ui.Output(
-			"Error listing pods: %s", clierrors.Humanize(err),
-			terminal.WithErrorStyle(),
-		)
+	s = sg.Add("Loading Helm chart...")
+	c, err := loader.Load(path)
+	if err != nil {
 		return nil, err
-	} else {
-		for _, pod := range podList.Items {
-			// patch the pod containers with the new i.config.serverImage
-			// Payload should be the updated server config image with the podspec
-			for j := range pod.Spec.Containers {
-				pod.Spec.Containers[j].Image = i.config.serverImage
-			}
-
-			jsonPayload, err := json.Marshal(pod)
-			if err != nil {
-				return nil, err
-			}
-
-			_, err = podClient.Patch(ctx, pod.Name, types.MergePatchType, jsonPayload, metav1.PatchOptions{})
-			if err != nil {
-				ui.Output(
-					"Error submitting patch to update container image: %s", clierrors.Humanize(err),
-					terminal.WithErrorStyle(),
-				)
-				return nil, err
-			}
-		}
 	}
+	s.Update("Helm chart loaded")
+	s.Status(terminal.StatusOK)
+	s.Done()
 
-	s.Update("Patch update sent to waypoint server pod(s)")
-
-	if waypointStatefulSet.Spec.UpdateStrategy.Type == "OnDelete" {
-		s.Update("Deleting pod to refresh image")
-		log.Info("Update Strategy is 'OnDelete', deleting pod to refresh image")
-
-		if podList, err := podClient.List(ctx, metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", serverName)}); err != nil {
-			ui.Output(
-				"Error listing pods: %s", clierrors.Humanize(err),
-				terminal.WithErrorStyle(),
-			)
-			return nil, err
-		} else {
-			for _, pod := range podList.Items {
-				if err := podClient.Delete(ctx, pod.Name, metav1.DeleteOptions{}); err != nil {
-					s.Update("Pod deletion failed", terminal.WithErrorStyle)
-					s.Done()
-					ui.Output(
-						"Error deleting pod %q: %s", pod.Name, clierrors.Humanize(err),
-						terminal.WithErrorStyle(),
-					)
-					return nil, err
-				}
-			}
-		}
-
-		log.Info("Pod(s) deleted, k8s will now restart waypoint server ", serverName)
-	} else if waypointStatefulSet.Spec.UpdateStrategy.Type == "RollingUpdate" {
-		log.Info("Update Strategy is 'RollingUpdate'; once the upgrade completes, you may need to restart the pod to update the server image")
-	} else {
-		log.Warn("Update Strategy is not recognized, so no action is taken", "UpdateStrategy",
-			waypointStatefulSet.Spec.UpdateStrategy.Type)
+	imageRef, err := dockerparser.Parse(i.config.serverImage)
+	if err != nil {
+		ui.Output("Error parsing image ref: %s", clierrors.Humanize(err), terminal.WithErrorStyle())
+		return nil, err
 	}
 
-	s.Update("Image set to update!")
-	s.Done()
-
-	s = sg.Add("Waiting for server to be ready...")
-	log.Info("waiting for waypoint server to become ready after image refresh")
+	values := map[string]interface{}{
+		"server": map[string]interface{}{
+			"enabled": true,
+			"image": map[string]interface{}{
+				"repository": imageRef.ShortName(),
```

Comment:
This should probably be `imageRef.Name()` and same for else where in this file.

Reply:
Although maybe not, I'm not sure how docker parses images but you might also need `imageRef.Repository()` too